pkgs/development/compilers/edk2/default.nix

```diff
@@ -1,16 +1,25 @@
-{ stdenv, fetchFromGitHub, fetchpatch, libuuid, python2, iasl }:
+{ stdenv, buildPackages, fetchFromGitHub, fetchpatch, libuuid, python2 }:
 
 let
-  pythonEnv = python2.withPackages(ps: [ps.tkinter]);
-
-targetArch = if stdenv.isi686 then
-  "IA32"
-else if stdenv.isx86_64 then
-  "X64"
-else if stdenv.isAarch64 then
-  "AARCH64"
-else
-  throw "Unsupported architecture";
+# Given a stdenv, returns the edk2-valid arch.
+envToArch = env:
+  if env.isi686 then
+    "IA32"
+  else if env.isx86_64 then
+    "X64"
+  else if env.isAarch64 then
+    "AARCH64"
+  else if env.isAarch32 then
+    "ARM"
+  else
+    throw "Unsupported architecture" 
+;
+
+buildPythonEnv = buildPackages.python2.withPackages(ps: [ps.tkinter]);
+pythonEnv = python2.withPackages(ps: [ps.tkinter]);
+
+targetArch = envToArch stdenv;
```

Comment:
this should use build/host/targetPlatform not stdenv

Reply:
The code it replaces is `if stdenv.isi686` and such. How do I decide which of those to prefer? I gather it would be targetPlatform, right?